### PR TITLE
Box filtering

### LINKED ISSUE #30: Improve detection box filtering strategies
We can improve detections using:
- Scene priors
- Background/Foreground subtraction and modeling

@alpha-carinae29 & @mrn-mln I suggest you take a look at this.

### PR DESCRIPTION
**Pull request by alpha-carinae29**
I implemented two box filtering strategy based on background subtraction and box prior distributions.
this will close #30

### CODE REVIEW

**Comment by mhejrati** [comment]
@alpha-carinae29 thanks a lot for this PR. Do you have any quantitative or qualitative results to compare performance of the detection before and after applying these filters?

**Comment by alpha-carinae29** [comment]
> @alpha-carinae29 thanks a lot for this PR. Do you have any quantitative or qualitative results to compare performance of the detection before and after applying these filters?

Yeah I have computed the mAP for last 1000 frames of Oxford Town Center dataset, however since the models are trained on this environment and have high accuracy and also there are relatively few number of frames for computing the prior distributions, there was not a distinguishable improvement in accuracy. We had just 0.2% improvement in mAP.

**Comment by mhejrati** [comment]
@alpha-carinae29 can you please resubmit this PR to the smart-social-distancing repo?

File: applications/smart-distancing/config-skeleton.ini [hunk 1]
```diff
@@ -19,6 +19,7 @@ MinScore: 0.25
 [PostProcessor]
 MaxTrackFrame: 5
 NMSThreshold: 0.98
+CalibrationFrames: 100
```
Comment (undefined-references) [inline applications/smart-distancing/config-skeleton.ini:22]: please provide explanation about post processing config parameters.

File: applications/smart-distancing/libs/core.py [hunk 2]
```diff
@@ -108,6 +113,8 @@ def calculate_distancing(self, objects_list):
         1. omitting large boxes by filtering boxes which are biger than the 1/4 of the size the image.
         2. omitting duplicated boxes by applying an auxilary non-maximum-suppression.
         3. apply a simple object tracker to make the detection more robust.
+        4. filter bounding boxes based on boxes prior distribution.
+        5. filter bounding boxes based on the amount of foreground pixels exist on the box.
```
Comment (undefined-references) [inline applications/smart-distancing/libs/core.py:118]: * Post processing step is getting complex and it's good to have a detailed documentation outside of code.
* I have an idea to develop a scoring algorithm for each bounding boxes based on these 5 steps criteria and detection score to omit extra bounding boxes in a unified way. each of steps add or reduce each bounding box acceptance score with a weighted score and bounding boxes with a score greater than a threshold will remain.

**Comment by mhejrati** [comment]
@alpha-carinae29 any updates on this?

**Comment by alpha-carinae29** [comment]
@mhejrati, can you review this?
I don't know what should I do. Thanks

**Comment by alpha-carinae29** [comment]
migrated to:
https://github.com/neuralet/smart-social-distancing/pull/11

**Pull request closed without merging**